packages/fastify/src/withClerkMiddleware.ts

```diff
@@ -1,39 +1,40 @@
-import { parse } from 'cookie';
+import { createIsomorphicRequest } from '@clerk/backend';
 import type { FastifyReply, FastifyRequest } from 'fastify';
 
 import { clerkClient } from './clerkClient';
 import * as constants from './constants';
 import type { ClerkFastifyOptions } from './types';
 import { getSingleValueFromArrayHeader } from './utils';
 
+const DUMMY_URL_BASE = 'http://clerk-dummy';
+
 export const withClerkMiddleware = (options: ClerkFastifyOptions) => {
   return async (req: FastifyRequest, reply: FastifyReply) => {
-    const cookies = parse(req.raw.headers.cookie || '');
     const secretKey = options.secretKey || constants.SECRET_KEY;
     const publishableKey = options.publishableKey || constants.PUBLISHABLE_KEY;
 
-    // get auth state, check if we need to return an interstitial
-    const cookieToken = cookies[constants.Cookies.Session];
-    const headerToken = req.headers['authorization']?.replace('Bearer ', '');
-
-    const forwardedPort = getSingleValueFromArrayHeader(req.headers['x-forwarded-port']);
-    const forwardedHost = getSingleValueFromArrayHeader(req.headers['x-forwarded-host']);
-
     const requestState = await clerkClient.authenticateRequest({
       ...options,
       secretKey,
       publishableKey,
       apiKey: constants.API_KEY,
       frontendApi: constants.FRONTEND_API,
-      cookieToken,
-      headerToken,
-      clientUat: cookies[constants.Cookies.ClientUat],
-      origin: req.headers['origin'] || undefined,
-      host: req.headers['host'] as string,
-      forwardedPort,
-      forwardedHost,
-      referrer: req.headers['referer'] || undefined,
-      userAgent: req.headers['user-agent'] || undefined,
+      request: createIsomorphicRequest((Request: any, Headers: any) => {
```

Comment:
❓ Why did you type these as `any` here?

These are the helpers you are passing to the caller of this function. I see you've defined this type:
```
type IsomorphicRequestOptions = (Request: Request, Headers: Headers) => Request;
```
meaning  that if you remove `any`, TS inference will work automatically giving you the correct type here.


```suggestion
      request: createIsomorphicRequest((Request, Headers) => {
```